### ISSUE TITLE
Memory issue

### ISSUE THREAD

**Issue by shock-wave007** [burhanrashid52/PhotoEditor#48]
![screenshot 24](https://user-images.githubusercontent.com/29371789/41193068-3e94032c-6c25-11e8-8701-199d3afc31f0.png)

if i change filter multiples time without "largerHeap="true" application crashes and as u can see in screenshot that it take upto 1.2Gb no phone, it some kind of memory leak or something.

**Comment by burhanrashid52** [comment]
We are using `GLSurfaceView` for image filtering which takes more resources to filter images. Thanks for the update I will look into it

**Comment by kustraslawomir** [comment]
@shock-wave007  @burhanrashid52 

**GLES20.glDeleteTextures(2, mTextures, 0)** before GLES20.glGenTextures(2, mTextures, 0) at ImageFilterView -> loadTextures() method resolved issue.

**Comment by IgorKlv** [comment]
@kustraslawomir I applied your solution, yet it has not helped to conserve memory and the App continually crashes after resuming filter selection.

**Comment by el-Knix** [comment]
Hey there did anyone manage to find a solution to this issue?

**Comment by mohammad1ta** [comment]
Hi,
No solution? :(

I have this issue to, I submitted this problem in [stackoverflow](https://stackoverflow.com/questions/54750859/memory-incremental-when-using-filter-image-photoeditor-sdk)

**Comment by mohammad1ta** [comment]
> @shock-wave007 @burhanrashid52
> 
> call **GLES20.glDeleteTextures(2, mTextures, 0)** before GLES20.glGenTextures(2, mTextures, 0) at loadTextures() method to resolve memory issue.

Thank you dear Kustra,
I tested your solution and helped me a few to control memory. ( after set filter, memory will be back to old step ).
But still its high memory usage...

**Comment by burhanrashid52** [comment]
The solution by @kustraslawomir worked @mohammad1ta ?

**Comment by mohammad1ta** [comment]
> The solution by @kustraslawomir worked @mohammad1ta ?

Yes, at the first time, When I use filter, memory will grow up from about 130MB to 232MB ( Because Graphics memory 0MB to 96MB ).
after second changing in filter, memory increases to 260MB and decrease back to 235MB.

**Comment by wifirevaithi** [comment]
hi, Is there is any permanent Solution For this Problem..? @burhanrashid52 @kustraslawomir

**Comment by maximusapp** [comment]
Your library work incorrect. If i change a few filters -> app crash without any error messages. this shit is not a library

**Comment by maximusapp** [comment]
You help me bro

**Comment by maximusapp** [comment]
> > @shock-wave007 @burhanrashid52
> > call **GLES20.glDeleteTextures(2, mTextures, 0)** before GLES20.glGenTextures(2, mTextures, 0) at loadTextures() method to resolve memory issue.
> 
> Thank you dear Kustra,
> I tested your solution and helped me a few to control memory. ( after set filter, memory will be back to old step ).
> But still its high memory usage...

You tip help me
